### PR TITLE
`RootView` actually shows the sticky footer

### PR DESCRIPTION
**Pull request by JayShortway**
The main change in this PR is using `safeAreaInset` in `RootView` to show the sticky footer. The other changes were required, in order to provide the sticky footer's `StackViewModel`.

### CODE REVIEW

File: RevenueCatUI/Templates/Components/PaywallComponentViewModel.swift [hunk 1]
```diff
@@ -30,6 +30,7 @@ enum PaywallComponentViewModel {
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 extension PaywallComponent {
 
+    // swiftlint:disable:next function_body_length
```
Comment (joshdholtz) [inline RevenueCatUI/Templates/Components/PaywallComponentViewModel.swift:33]: 🫠🫠🫠

File: RevenueCatUI/Templates/Components/Root/RootView.swift [hunk 2]
```diff
@@ -28,7 +28,14 @@ struct RootView: View {
...
     var body: some View {
-        StackComponentView(viewModel: viewModel.stackViewModel, onDismiss: onDismiss)
+        ScrollView {
+            StackComponentView(viewModel: viewModel.stackViewModel, onDismiss: onDismiss)
+        }.applyIfLet(viewModel.stickyFooterViewModel) { stackView, stickyFooterViewModel in
+            stackView.safeAreaInset(edge: .bottom) {
+                StackComponentView(viewModel: stickyFooterViewModel.stackViewModel, onDismiss: onDismiss)
+            }
+
+        }
```
Comment (joshdholtz) [inline RevenueCatUI/Templates/Components/Root/RootView.swift:38]: The simplicity here in the footer view is is 😍

**Review by joshdholtz — APPROVED** [review]
:shipit: :shipit: :shipit:

File: RevenueCatUI/Templates/Components/Root/RootView.swift (hunk 2 above)
Comment (JayShortway) [reply RevenueCatUI/Templates/Components/Root/RootView.swift:38]: Yea it's pretty great! I hope we can keep it that way haha

**Pull request merged**